src/db.c

```diff
@@ -409,18 +400,13 @@ int dbGenericDelete(redisDb *db, robj *key, int async, int flags) {
         if (async) {
             /* Because of dbUnshareStringValue, the val in de may change. */
             freeObjAsync(key, dictGetVal(de), db->id);
-            dictSetVal(d, de, NULL);
+            daDictSetVal(db->keys, slot, de, NULL);
         }
         /* Deleting an entry from the expires dict will not free the sds of
          * the key, because it is shared with the main dictionary. */
-        dict *dv = daGetDict(db->volatile_keys, slot);
-        if (dictSize(dv) > 0) {
```

Comment:
i wonder why we had that dictSize call? is it completely excessive? or was there some hidden reason?
can you check the blame log?

Reply:
it dates back to before 2010, a lot of code movements in these early years, I gave up after 5-6 such commits

my guess is that it was just an attempt to save some CPU cycles for hashing the keyname in case there are no volatile keys? perhaps in case a user doesn't even work with volatile keys at all?

should i revive that check? or better yes, early-exit inside dictDelete if the dict is empty?

Reply:
dictFind has an early abort if the dict is empty. anyway, fine. let's dismiss it.

Reply:
ok, so i'm leaving the code as-is and resolving this comment